frontend/src/app/profile-edit/index.tsx

```diff
@@ -0,0 +1,342 @@
+      : updateError;
+
+  const getFieldError = field =>
+    (updateData &&
+      updateData.update.__typename === "UpdateErrors" &&
+      updateData.update[field].join(", ")) ||
+    "";
+
+  const toTileCase = word => word.charAt(0).toUpperCase() + word.slice(1);
+
+  const errorsFields = {};
+  FORM_FIELDS.forEach(field => {
+    const validationField = "validation" + toTileCase(field);
+    errorsFields[field] = getFieldError(validationField);
+  });
+  // endregion
+
+  // endregion
+
+  const hangleUserChange = ({ target }) => {
+    formState.setField(target.id, target.value);
+  };
+
+  useEffect(() => {
+    // Update the document title using the browser API
+    console.log(
+      "Effect: formState.values: " + JSON.stringify(formState.values),
+    );
+  });
+
+  return (
+    <>
+      <h1>
+        <FormattedMessage id="profile.header"/>
+      </h1>
+
+      {loading && "Loading..."}
+      {!loading && (
+        <Form onSubmit={onFormSubmit} method="post">
+          {errorMessage && <Alert type="error">{errorMessage}</Alert>}
+          <Row paddingBottom={ROW_PADDING} paddingTop={BUTTON_PADDING}>
+            <Column columnWidth={COLUMN_WIDTH}>
+              <LayoutSet>
+                <Card>
+                  <h3>
+                    <FormattedMessage id="profile.edit.personalHeader"/>
+                  </h3>
+
+                  <InputWrapper text={errorsFields.firstName} isRequired={true}>
+                    <InputField
+                      inputProps={{
+                        id: "firstName",
+                        ...text("firstName"),
+                      }}
+                      a11yId="firstName"
+                      // fyi: <Label> doesn't have isRequired property, doesn't show
+                      // "*" to indicate that the field is required
+                      label={
+                        <FormattedMessage id="profile.firstName">
+                          {msg => <b>{msg}</b>}
+                        </FormattedMessage>
+                      }
+                      onChange={hangleUserChange}
+                      isRequired={true}
+                    />
+                  </InputWrapper>
+
+                  <InputWrapper text={errorsFields.lastName} isRequired={true}>
+                    <InputField
+                      inputProps={{
+                        id: "lastName",
+                        ...text("lastName"),
+                      }}
+                      a11yId="lastName"
+                      label={
+                        <FormattedMessage id="profile.lastName">
+                          {msg => <b>{msg}</b>}
+                        </FormattedMessage>
+                      }
+                      onChange={hangleUserChange}
+                    />
+                  </InputWrapper>
+
+                  <InputWrapper text={errorsFields.gender} isRequired={true}>
+                    <RadioGroupField
+                      {...radio("gender",
```

Comment:
The components `radio`, `select`, `raw` of react-use-form-state they wrapped the field with a different id, 
if I use hangleUserChange it save the change in a different random field.. so I re-do the update here manually... 
I don't like very much...